modules/sqs-queue/remote-state.tf

```
@@ -0,0 +1,11 @@
+
+module "dead_letter_sqs_remote_state" {
+  count = var.dead_letter_sqs_component_name != null ? 1 : 0
```

Maybe to be safe with `""` we should do

```suggestion
  count = try(length(var.dead_letter_sqs_component_name), 0) > 0 ? 1 : 0
```

---

I think this is safe enough, the variable is defaulted to null, so someone would have to be explicitly trying to set the component name to `""`.